docs/pull-requests.rst

```diff
@@ -0,0 +1,73 @@
+Preview Documentation from Pull Requests
+========================================
+
+Read the Docs allows to build and preview your documentation from pull/merge requests.
+You can enable it by: 
+
+#. Go to your project dashboard
+#. Go to :guilabel:`Admin > Advanced settings`
+#. Enable the :guilabel:`Build pull requests for this project` option
+#. Click on :guilabel:`Save`
+
+Features
+--------
+
+- **Build on Pull/Merge Request Events:** We create and build a new version when a pull/merge request is open,
+  and when a new commit has been pushed.
+
+- **Warning Banner:** A warning banner is shown at the top of the documentation
+  to let users know that this isn't the main documentation for the project.
+
+  .. note:: This feature is available only for :doc:`Sphinx projects </intro/getting-started-with-sphinx>`.
+
+- **Build Status Report:** When a build is triggered, a build pending notification is send with a link to the build log,
+  and when the build is done we send a success notification with the link to the preview or a failure notification with a link to the build log.
+
+.. figure:: /_static/images/github-build-status-reporting.gif
+   :align: center
+   :alt: GitHub Build Status Reporting for Pull Requests.
+   :figwidth: 80%
+   :target: ../_static/images/guides/github-build-status-reporting.gif
+
+   GitHub build status reporting
+
+Privacy levels
+--------------
+
+.. note::
+
+   Privacy levels are only supported on :doc:`/commercial/index`.
+
+All docs built from a pull/merge requests are private by default.
+Currently, this can't be changed, but we are planning to support this.
+
+Limitations
+-----------
+
+- Auto-builds for pull/merge requests have
+  :doc:`the same limitations as regular documentation builds </builds>`.
+- Only available for GitHub and GitLab.
+- Additional formats like PDF and Epub aren't build.
+- Searches will default to the default experience for your tool.
+  This is a feature we plan to add,
+  but don't want to overwhelm our search indexes used in production.
```

Comment:
Is this true? I thought search just didn't work :/

Reply:
It should just work, as they are regular versions, we can enable it by removing this line

https://github.com/readthedocs/readthedocs.org/blob/74df71db645222926659c265c33b9abf577a9972/readthedocs/projects/tasks.py#L1318-L1320

Reply:
now with the custom options for search I think this is useful, so users can test the results.

Reply:
Yea, it will just destroy our search indexes. I don't want to break prod search by indexing PR's, so we probably want a different cluster for it, if nothing else.